examples/todo/packages/contracts/src/systems/TodoSystem.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import { console } from "forge-std/console.sol";
+import { System } from "@latticexyz/world/src/System.sol";
+
+import { TodoItem, TodoItemData, Owned } from "../codegen/Tables.sol";
+
+function addressToEntity(address a) pure returns (bytes32) {
+  return bytes32(uint256(uint160((a))));
+}
+
+contract TodoSystem is System {
+  uint256 totalTodos;
+
+  function add(string memory body) public {
+    bytes32 user = addressToEntity(_msgSender());
+    bytes32 todoKey = keccak256(abi.encodePacked(user, totalTodos++));
```

Comment:
I thought state inside a system was a MUD antipattern?

Could use we this module instead? https://github.com/latticexyz/mud/pull/552